public/apps/configuration/panels/role-edit/role-edit.tsx

```diff
@@ -96,27 +123,58 @@ export function RoleEdit(props: RoleEditDeps) {
         const actionGroupsObject = await fetchActionGroups(props.coreStart.http);
         setActionGroups(Object.keys(actionGroupsObject));
       } catch (e) {
-        // TODO: show user friendly error message
-        console.log(e);
+        addToast({
+          id: 'actionGroup',
+          color: 'danger',
+          title: 'Failed to load data',
+          text:
+            'Failed to load data. You may refresh the page to retry or see browser console for more information.',
```

Comment:
This block has repeated 3 times. Probably better to create a function addToast which only takes id as input. I will leave it up to you to decide.

Reply:
Good idea, will do.